add spiderpool metrics markdown (#672)
spidernet-io/spiderpool: pkg/metric/README.md

# Metrics

Spiderpool can be configured to serve [Opentelemetry](https://opentelemetry.io/) metrics.
Spiderpool metrics provide the insight of Spiderpool Agent and Spiderpool Controller.

## Get Started

### Enable Metric support

> Firstly, please ensure you have installed the spiderpool and configure the CNI file, refer [install](./install.md) for details

Reviewer: ```suggestion
Firstly, please ensure you have installed the spiderpool and configure the CNI file, refer to [install](./install.md) for details
```

> Check daemonset `spiderpool-agent` environment `SPIDERPOOL_ENABLED_METRIC` whether is already set to `true` or not.

Reviewer: ```suggestion
Check the environment variable `SPIDERPOOL_ENABLED_METRIC` of the daemonset `spiderpool-agent`  for whether it is already set to `true` or not.
```

> Check deployment `spiderpool-controller` environment `SPIDERPOOL_ENABLED_METRIC` whether is already set to `true` or not.

Reviewer: ```suggestion
Check the environment variable `SPIDERPOOL_ENABLED_METRIC` of the deployment `spiderpool-controller` for whether it is already set to `true` or not.
```

```shell
kubectl -n kube-system get daemonset spiderpool-agent -o yaml
------
kubectl -n kube-system get deployment spiderpool-controller -o yaml
```

> You can set one of them or both of them to `true`.

Reviewer: ```suggestion
You can set one or both of them to `true`.
```

> For example, let's enable spiderpool agent metrics, just execute `helm upgrade --set spiderpoolAgent.prometheus.enabled=true`.

Reviewer: ```suggestion
For example, let's enable spiderpool agent metrics by running `helm upgrade --set spiderpoolAgent.prometheus.enabled=true`.
```

## Metric reference

### Spiderpool Agent

Spiderpool agent exports some metrics related with IPAM allocation and deallocation. Currently, those include:

| Name                                          | description                                                                                                                    |
|-----------------------------------------------|--------------------------------------------------------------------------------------------------------------------------------|

> | ipam_allocation_total_counts                  | Count of the number of Spiderpool Agent received the IPAM allocation requests, prometheus type: counter                        |

Reviewer: ```suggestion
| ipam_allocation_total_counts                  | Count of the number of Spiderpool Agents that received the IPAM allocation requests, prometheus type: counter                        |
```

| ipam_allocation_failure_counts                | Number of Spiderpool Agent IPAM allocation failures, prometheus type: counter                                                  |

> | ipam_allocation_average_duration_seconds      | The average time duration of total Spiderpool Agent allocation process, prometheus type: gauge                                 |

Reviewer: ```suggestion
| ipam_allocation_average_duration_seconds      | The average duration of all Spiderpool Agent allocation processes, prometheus type: gauge                                 |
```

> | ipam_allocation_max_duration_seconds          | The maximum time duration of Spiderpool Agent allocation process (per-process), prometheus type: gauge                         |

Reviewer: ```suggestion
| ipam_allocation_max_duration_seconds          | The maximum duration of Spiderpool Agent allocation process (per-process), prometheus type: gauge                         |
```
Need to replace `time duration` with `duration`.
